test/libsolidity/SolidityNameAndTypeResolution.cpp

```diff
@@ -1260,6 +1260,49 @@ BOOST_AUTO_TEST_CASE(multiple_events_argument_clash)
 	BOOST_CHECK(success(text));
 }
 
+BOOST_AUTO_TEST_CASE(function_function_clash)
```

Comment:
@chriseth please decide if this source should compile or not.  I can give arguments for both choices.